leetcode/00925-LongPressedName/liz.py

```diff
@@ -0,0 +1,26 @@
+class Solution:
+    def isLongPressedName(self, name: str, typed: str) -> bool:
+        
+        n, t = 0, 0
+        len_name, len_typed = len(name) - 1, len(typed) - 1
+        
+        while t <= len_typed:
+            
+            if name[n] == typed[t]:
+                if n < len_name and t < len_typed:
+                    n += 1
+                    
+                t += 1
+            
+            else:
+                if n == len_name and t == len_typed:

```

Comment:
do we need `t == len_typed`? the while loops terminates when `t <= len_typed`

Reply:
this should catch cases like this:
`name = "abc"`
`typed = "abb"` 
where `name[2]` (c) is different from `typed[2]` (b). without this condition, the code will move to lines19-20 and match `typed[2]` (b) to the previous letter `typed[1]` (b) and return true